### ISSUE TITLE
Rule setting suggestion: "allow-invoked" with no-mixed-requires [$25]

### ISSUE THREAD

**Issue by indexzero** [eslint/eslint#4011]
Read through open and closed issues about `no-mixed-requires` and didn't see anything like this. Would allow syntax like this:

``` js
var async = require('async'),
    debug = require('diagnostics')('my-module'),
    eslint = require('eslint');
```

But still disallow all mixed constants,etc. Currently the work around for this is to simply put the immediately computed result in a separate `var` statement:

 ``` js
var async = require('async'),
    eslint = require('eslint');

var debug = require('diagnostics')('my-module');
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
I think we need to think through the name of the option a bit, but seems like a good idea. Do you want to submit a pull request?

**Comment by indexzero** [comment]
I'll take a stab at it, could be tricky though.

**Comment by rpatil26** [comment]
@indexzero - Are you looking at this? If not, I was thinking to give it a try. The option name I am thinking of is "allowInvocation". The options have been camelCase and let's keep it consistent with that practice.

**Comment by rpatil26** [comment]
@indexzero - One more question. With this new option true, would following be considered a problem? If so, then it would be tricky. 
```js
var async = require('async'),
    debug = require('diagnostics').someFunction('my-module'),
    eslint = require('eslint');
```

**Comment by nzakas** [comment]
I'd shorten it to `allowCall` and would not allow that code.

**Comment by rpatil26** [comment]
Ok,  I'll name it `allowCall`. 

Let me explore it more to allow ```require('module')('nested'),``` but not ```require('module').someFn('param')```

**Comment by indexzero** [comment]
Great work @rpatil26.

**Comment by rpatil26** [comment]
Thanks @indexzero. Looking forward to more contributions.

**Comment by nzakas** [comment]
@rpatil26 be sure to claim your bounty for this issue.
